subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -174,14 +226,21 @@ void bl_boot(const struct fw_info *fw_info)
 		"   b   clear\n"
 		"out:\n"
 		"   dsb\n"
+#ifndef CONFIG_SB_DISABLE_SELF_R_X
 		/* Jump to reset vector of an app */
 		"   bx r0\n"
+#endif /* CONFIG_SB_DISABLE_SELF_R_X */
 		:
 		: "r" (vector_table[1]), "i" (CONFIG_SRAM_BASE_ADDRESS),
 		  "i" (CONFIG_SRAM_SIZE * 1024), "i" (0)
 		: "r0", "r1", "r2", "r3", "memory"
 	);
-#else
+#endif /* CONFIG_SB_CLEANUP_RAM */
+
+#if defined(CONFIG_SB_DISABLE_SELF_R_X)
+	relocate_bootconf_disable_r_x_and_jump()(vector_table[1]);
```

Comment:
don't think this will work, at this point the RAM is blank, and you are returning to C-land where it expects to have a fully working stack pointer including all data on stack. Would just make more sense to have one asm block with #ifdefs in it to select which parts are enabled or not, without calling C functions in-between